src/components/editMeasure/EditMeasureNav.tsx

```diff
@@ -12,19 +12,23 @@ export interface RouteHandlerState {
   canTravel: boolean;
   pendingRoute: string;
 }
+import useTestCaseServiceApi from "../../api/useTestCaseServiceApi";
 interface PropTypes {
   isActive?: boolean;
 }
 
 const MenuItemContainer = tw.ul`bg-transparent flex px-8`;
+
 const MenuItem = styled.li((props: PropTypes) => [
   tw`mr-1 text-white bg-[rgba(0, 32, 64, 0.5)] rounded-t-md hover:bg-[rgba(0,6,13, 0.5)]`,
   props.isActive && tw`bg-slate text-slate-90 font-medium hover:bg-slate`,
 ]);
 
 const EditMeasureNav = ({ isQDM }) => {
+  const [testCaseLength, setTestCaseLength] = useState<any>(0);
   const { pathname } = useLocation();
   let navigate = useNavigate();
+  const testCaseService = useRef(useTestCaseServiceApi());
   const { id } = useParams<{
```

Comment:
This is not a part of your work, but it will very helpful for future, if we can rename "id" to `measureId` or `currentMeasureId` which provides a lot of context than just saying "id"